parent/pom.xml

```diff
@@ -71,7 +71,6 @@
       ${camunda.osgi.import.before.defaults},
       ${camunda.osgi.import.defaults},
       ${camunda.osgi.import.additional},
-      org.camunda.*;${camunda.osgi.import.camunda.version},
```

Comment:
Could you shortly detail why you would want to remove that part?

Reply:
This is unnecessary import package because it has been handled by * in this property 

> camunda.osgi.import.pkg

When you have that import in camunda.osgi.import.pkg, it will cause an OutOfMemoryError

```
[ERROR] Java heap space -> [Help 1]
java.lang.OutOfMemoryError: Java heap space
    at java.util.Arrays.copyOf (Arrays.java:3332)
    at java.lang.AbstractStringBuilder.ensureCapacityInternal (AbstractStringBuilder.java:124)
    at java.lang.AbstractStringBuilder.append (AbstractStringBuilder.java:448)
    at java.lang.StringBuilder.append (StringBuilder.java:136)
    at java.lang.StringBuilder.append (StringBuilder.java:76)
    at java.lang.AbstractStringBuilder.append (AbstractStringBuilder.java:484)
    at java.lang.StringBuilder.append (StringBuilder.java:166)
    at java.lang.StringBuilder.append (StringBuilder.java:76)
    at aQute.bnd.osgi.Processor.report (Processor.java:1678)
    at aQute.bnd.osgi.Processor.toString (Processor.java:2114)
    at java.util.Formatter$FormatSpecifier.printString (Formatter.java:2886)
    at java.util.Formatter$FormatSpecifier.print (Formatter.java:2763)
    at java.util.Formatter.format (Formatter.java:2520)
    at java.util.Formatter.format (Formatter.java:2455)
    at java.lang.String.format (String.java:2940)
    at aQute.bnd.osgi.Processor.formatArrays (Processor.java:1784)
    at aQute.bnd.osgi.Processor.warning (Processor.java:169)
    at aQute.bnd.osgi.Macro.replace (Macro.java:217)
    at aQute.bnd.osgi.Macro.process (Macro.java:62)
    at aQute.bnd.osgi.Macro.process (Macro.java:78)
    at aQute.bnd.osgi.Macro.process (Macro.java:48)
    at aQute.bnd.osgi.Macro.process (Macro.java:43)
    at aQute.bnd.osgi.Macro.process (Macro.java:1139)
    at aQute.bnd.osgi.Analyzer.applyVersionPolicy (Analyzer.java:1644)
    at aQute.bnd.osgi.Analyzer.augmentImports (Analyzer.java:1602)
    at aQute.bnd.osgi.Analyzer.analyze (Analyzer.java:268)
    at aQute.bnd.osgi.Builder.analyze (Builder.java:343)
    at aQute.bnd.osgi.Analyzer.calcManifest (Analyzer.java:616)
    at aQute.bnd.osgi.Builder.build (Builder.java:80)
    at org.apache.felix.bundleplugin.ManifestPlugin.getAnalyzer (ManifestPlugin.java:208)
    at org.apache.felix.bundleplugin.ManifestPlugin.getManifest (ManifestPlugin.java:114)
    at org.apache.felix.bundleplugin.ManifestPlugin.execute (ManifestPlugin.java:69)
```

Reply:
That sounds reasonable, thanks for explaining! 👍

Reply:
@tmetzke ,
Yes you're right. By default maven bundle plugin will calculate the range of import package. Please refer to this link https://bnd.bndtools.org/heads/import_package.html.

Let's say the version project is 7.15, the plugin will generate import package with version [7.15, 8). Unless you want to support lower version than 7.15, I don't see a need to explicitly specify the range version.

Reply:
So, a Camunda artifact will then use the newest version of other Camunda artifacts it can find that are greater than the minimum but not yet 8.x. This would imply that a 7.15 artifact could use a 7.16 or 7.17 artifact if it can find one, right? As far as I understand or current description, we ensure the artifacts use tested and compatible versions of other artifacts. This would be removed with the proposed change, wouldn't it?

Reply:
The OSGi default is importing with range that agree to semantic versioning. Previously it force the exact minor version of camunda that does not allow to get update from one bundle with the desiderate feature/fix. Keep in mind that the restriction you force in the manifest is valid only in OSGi but in any other J2EE enviroment you can mix the bundle you want.
The properties system you used if very very complicated to read and also to handle (maybe the reason before was embedded), the resolution of that specific property causes the OOM in this maven plugin version and different error (property is never resolved to their final value) with most recent plugin version 5.1.1.
What could we do in this case?

Reply:
Hey Nikolas, thanks for that input. If you are fine with taking care of the versions yourself, removing this will work for us. You are right with that option being present in other environments like in J2EE use cases. Plus, if this poses an issue currently, we are also happy to resolve it with this move.